<launch>
  <arg name="map_name" default="sztaki_real" doc="Name of the map yaml file without extension, contained in the assets folder." />
  <arg name="bag" default="take01" />
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]" />

  <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
    <arg name="model" value="$(arg model)" />
  </include>

  <node name="map_server" pkg="map_server" type="map_server" output="screen" args="$(find dhf_loc)/assets/$(arg map_name).yaml" />

  <include file="$(find turtlebot3_navigation)/launch/amcl.launch">
    <arg name="initial_pose_x" value="0.05" />
    <arg name="initial_pose_y" value="0.075" />
    <arg name="initial_pose_a" value="0" />
  </include>

  <node pkg="dhf_loc" type="topicexporter_node.py" name="topicexporter_node" output="screen">
    <param name="pkg_name" value="dhf_loc" />
    <param name="export_file_name" value="amcl_$(arg bag)" />
    <param name="amcl_topic" value="/amcl_pose" />
    <param name="amcl_comptime_topic" value="/amcl_comptime" />
    <param name="detection_tolerance" value="0.1" />


  </node>

  <node name="rosbag" pkg="rosbag" type="play" output="screen" args="$(find dhf_loc)/assets/bags/$(arg bag).bag -q" />


  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find turtlebot3_navigation)/rviz/turtlebot3_navigation.rviz" />
</launch>